packages/knip/src/plugins/aws-cdk/index.ts

```diff
@@ -0,0 +1,38 @@
+import { toDependency } from 'src/util/input.js';
+import type { IsPluginEnabled, Plugin, ResolveConfig } from '../../types/config.js';
+import { compact } from '../../util/array.js';
+import { hasDependency } from '../../util/plugin.js';
+import type { AwsCdkConfig } from './types.ts';
+
+// https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib-readme.html
+
+const title = 'aws-cdk';
+
+const enablers = ['aws-cdk'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const config: string[] = ["cdk.json"];
+
+const resolveConfig: ResolveConfig<AwsCdkConfig> = async config => {
+  if (!config) return [];
+
+  const app = config.app.split(" ")[0];
+  const watch = config.watch?.include ?? [];
+  const context = Object.keys(config.context ?? {}).map(key => key.split("/")[0]) ?? [];
+  return compact([app, ...watch, ...context]).map(id => toDependency(id));
```

Comment:
I like that we're using `compact` here and already convert string to so-called "inputs" using `toDependency`. However, even though the `context` items are such dependencies indeed, returned items could also be entry points. You'll see what I mean when using `getInputsFromScripts` - its return value could contain various types of inputs.

Reply:
Are you meaning that I should be checking for the type of `id` in the `map` function? I saw typescript complaining about mismatched type and used something I saw somewhere else in the codebase:

```typescript
compact([...app, ...watch, ...context]).map(id => (typeof id === 'string' ? toDependency(id) : id));
```

Reply:
Yes, something like that, e.g. the return value of `getInputsFromScripts` is input, not strings.